File: backend/tools/write_file.py
from workspace import write_file as _write_file

async def write_file(path: str, content: str):
    _write_file(path, content)
    preview = content[:500] + ("..." if len(content) > 500 else "")
    return {"status": "saved", "path": path, "preview": preview}

File: frontend/src/tools/writeFile.ts
import type { Tool } from "./types"
import { callWorkspaceApi } from "./backendApi"

export const writeFileTool: Tool = {
  name: "write_file",

  description: "Write content to a workspace file",

  parameters: {
    type: "object",
    properties: {
      path: { type: "string" },
      content: { type: "string" }
    },
    required: ["path", "content"]
  },

  async execute({ path, content }) {
    await callWorkspaceApi("/workspace/write", { path, content })
    return { success: true }
  }
}